templates/etc/puppetserver/ca.cfg.erb

```diff
@@ -2,6 +2,9 @@
 # Any changes will be removed on the next run
 <% if @enable_ca -%>
 puppetlabs.services.ca.certificate-authority-service/certificate-authority-service
+<%  if Gem::Version.new(@_server_version) >= Gem::Version.new('5.1.0') -%>
```

Comment:
Logic and knowledge should not be in templates

Reply:
@jeefberkey This really doesn't seem better in any way:

In `pupmod::master`

```ruby
if versioncmp($_server_version, '5.4.1') < 0 {
  $_server_version_less_than_5_4_1 = true
}
else {
  $_server_version_less_than_5_4_1 = false
}
```

Random Template

```erb
<% if $pupmod::master::server_version_less_than_5_4_1 -%>
stuff
<% end -%>
```
I mean, how far down this rabbit hole do we go as new versions come out?

* `$_puppet_server_between_5_4_1_and_less_than_6`
* `$_puppet_server_greater_than_6_and_less_than_7`
* etc...

I'm switching all templates to EPP at this point but I think that logic should go in templates when it makes sense. Otherwise, we're going to have a right mess of random variables like the situation above and/or we'll be piecing together files in ways that are just going to make it complicated.

Absolutely happy to keep this going for debate but, after trying it, I don't think the trade-off is worth it given that it's literally only for templates and just turns the logic into a variable with the same name as the logic. Choosing between two things that are ugly and confusing, I'd like to presently stick with one that confines the confusion to the location that it actually affects and doesn't spread it around different files without (from this experiment) discernible benefit.